src/main/java/org/janelia/saalfeldlab/paintera/meshes/MeshManagerWithAssignmentForSegments.java

```diff
@@ -244,7 +246,8 @@ private void removeMeshes(final Map<Long, MeshGenerator<TLongHashSet>> toBeRemov
 		root.getChildren().setAll(existingGroups);
 
 		// unbind() for each mesh here takes way too long for some reason. Do it on a separate thread to avoid app freezing.
-		new Thread(() -> toBeRemoved.values().forEach(MeshGenerator::unbind)).start();
+		// TODO this should probably not create new threads, revisit!!
+		new Thread(() -> toBeRemoved.values().forEach(mg -> mg.meshSettingsProperty().unbind())).start();
```

Comment:
I addressed this in #267 by simply removing this line of code. JavaFX's [bind()](https://docs.oracle.com/javase/8/javafx/api/javafx/beans/property/Property.html#bind-javafx.beans.value.ObservableValue-) is said to be implemented using weak listeners (in contrast to [addListener()](https://docs.oracle.com/javafx/2/api/javafx/beans/value/ObservableValue.html#addListener(javafx.beans.value.ChangeListener)) that uses strong references), so it seems that explicit unbinding here is not necessary. Hopefully this will not introduce more memory leaks :grimacing:

Reply:
Sorry, disregard my previous comment -- I didn't realize that now it would remove an actual listener and not just unbind the properties